src/transformState.js

```diff
@@ -5,7 +5,42 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        const { extraData } = action;
+
+        for (const key in extraData) {
+          state[key] = extraData[key];
+        }
+        break;
+      }
+
+      case 'removeProperties': {
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      }
+
+      default: {
```

Comment:
Throwing an error for an unknown action type could stop the execution of the function prematurely. If this behavior is not intended, consider logging a warning or ignoring the unknown type instead.